README.md

```diff
@@ -4,364 +4,565 @@ The purpose of the ODM is to support wastewater-based surveillance and epidemiol
 
-### 2. Selecting an Entity
+### 2. Rules and Modes
 
-The second step involves selecting the parts of the ODM data model or entities that the rule applies to. The entities that can be selected are:
+After defining the unique ID for a rule, the next step is to determine the `mode` of a rule. There are four possible values for the `mode` column:
 
-* Data contained in a table
-* Data contained in column(s) of table(s)
+1. `select`: This indicates that the effect of this rule will be to select the tables and columns for inclusion in the output shareable data. It also means that the `key` and `operator` columns do not need to be defined for this rule.
+2. `filter`: This is used for rules that will filter the shareble data output rows based on row values. The full rule will require the `key` and `operator` columns to be fully specified.
+3. `group`: This defines activities for a rule that groups or combines rules together for execution. The full rule will require the `key` and `operator` columns to be fully specified.
+4. `share`: This defines a rule that specifies the `organizationID` or `contactID` with which an output will be shared, as well as the rules to apply to generate the specific output data. The full rule will require the `key` column, but not the `operator` column, to be fully specified.
 
-This step uses two columns, `tableName` and `variableName`. The `tableName` column can specify the name(s) of the table this rules applies to and the `variableName` column can specify the name(s) of the columns this rule applies to.
+Generally, the bulk of a sharing csv will be composed of `filter` and `select` rules, with a few `group` rules, and the final `share` rules at the very end. Rules should also be written and specified in this same order.
 
-Some examples are given below:
+### 3. Selecting an Entity
+
+In order to generate an intelligible output dataset, several `select` and `filter` rules will need to first be stacked and applied. This step involves selecting the parts of the PHES-ODM or entities within the model. The entities that can be selected are:
 
-1. Selecting the `type` column in the `Sample` table
+-   Data contained in a table
+-   Data contained in column(s) of table(s)
+-   Data contained in row(s) of table(s)
 
-    | ruleId | tableName  | variableName |
-    | ------ | ---------- | ------------ |
-    | 1      | Sample     | type         |
+This step uses four columns, `table`, `mode`, `key`, and/or `value`. The `table` column specifies the name(s) of the table(s) to which this rule applies. The `mode` column specifies the action of a rule the name(s) of the column(s) to be included in the shared data output as specified by this rule. For rules that select entities, the `filter` and `select` modes will be used.
```

Comment:
Is something missing from this sentence? Its formatted weirdly. I think it needs to say that the `key` column specifies the `value` column specifies the name(s) of the column(s)....

Reply:
Will fix this too!